tools/api-markdown-documenter/src/index.ts

```diff
@@ -38,9 +38,44 @@ export {
 	transformTsdoc,
 } from "./api-item-transforms/index.js";
 
-// We want to make sure the entirety of this domain is accessible.
-// eslint-disable-next-line no-restricted-syntax
-export * from "./documentation-domain/index.js";
+export {
```

Comment:
Note for reviewers: Required to avoid leaking `createNodesFromPlainText` into the API surface. But since this aligns with our linter requirements anyways, I guess this is an improvement.